packages/components/src/Menu/Menu.test.tsx

```diff
@@ -33,74 +32,126 @@ it("renders a Menu", () => {
-                label: "Text Message",
-                icon: "sms",
-              },
-              {
-                label: "Email",
-                icon: "email",
-              },
-            ],
-          },
-        ]}
-      />,
-    )
-    .toJSON();
-  expect(tree).toMatchSnapshot();
-});
+  describe("when clicking on the activator", () => {
+    it("should open the menu", async () => {
+      const header = "Mark as...";
+      const actionLabel = "Awaiting Response";
+      const clickHandler = jest.fn();
+      const actions = [
+        {
+          header: header,
+          actions: [{ label: actionLabel, onClick: clickHandler }],
+        },
+      ];
+      const { getByRole } = render(<Menu items={actions} />);
+      fireEvent.click(getByRole("button"));
+      await waitFor(() => {
+        expect(getByRole("menuitem")).toBeInTheDocument();
+      });
+    });
+  });
 
-test("it should open and close the menu", async () => {
-  const header = "Mark as...";
-  const actionLabel = "Awaiting Response";
-  const clickHandler = jest.fn();
-  const actions = [
-    {
-      header: header,
-      actions: [{ label: actionLabel, onClick: clickHandler }],
-    },
-  ];
+  describe("when clicking on menu item", () => {
+    it("should close the menu and trigger onClick", async () => {
+      const header = "Mark as...";
+      const actionLabel = "Awaiting Response";
+      const clickHandler = jest.fn();
+      const actions = [
+        {
+          header: header,
+          actions: [{ label: actionLabel, onClick: clickHandler }],
+        },
+      ];
+      const { getByRole, queryByRole } = render(<Menu items={actions} />);
 
-  const { getByRole, queryAllByText } = render(<Menu items={actions} />);
+      fireEvent.click(getByRole("button"));
+      fireEvent.click(getByRole("menuitem"));
+      expect(clickHandler).toHaveBeenCalledTimes(1);
+      await waitFor(() => {
+        expect(queryByRole("menuitem")).not.toBeInTheDocument();
+      });
+    });
+  });
 
-  fireEvent.click(getByRole("button"));
-  expect(queryAllByText(actionLabel).length).toBe(1);
+  describe("when menu is opened and escape is pressed", () => {
+    const actionLabel = "Text Message";
+    it("should close the menu", async () => {
+      const { getByRole, queryByRole, container } = render(
+        <Menu
+          items={[
+            {
+              header: "Send as...",
+              actions: [
+                {
+                  label: actionLabel,
+                  icon: "sms",
+                },
+                {
+                  label: "Email",
+                  icon: "email",
+                },
+              ],
+            },
+          ]}
+        />,
+      );
 
-  fireEvent.click(getByRole("menuitem"));
-  expect(clickHandler).toHaveBeenCalledTimes(1);
+      fireEvent.click(getByRole("button"));
+      fireEvent.keyDown(container, {
+        key: "Escape",
+        code: "Escape",
+        ctrlKey: true,
```

Comment:
interesting. y u testing with the `ctrlKey` pressed?

Reply:
I had copied the fire event from another test and didn't notice the ctrl key 🤭 

removed it in 99e07af as it's not needed for the test